test/org/opensolaris/opengrok/configuration/messages/ProjectMessageTest.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+ /*
+ * Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.configuration.messages;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.opensolaris.opengrok.condition.ConditionalRun;
+import org.opensolaris.opengrok.condition.ConditionalRunRule;
+import org.opensolaris.opengrok.condition.RepositoryInstalled;
+import org.opensolaris.opengrok.configuration.Project;
+import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
+import org.opensolaris.opengrok.history.HistoryGuru;
+import org.opensolaris.opengrok.history.MercurialRepositoryTest;
+import org.opensolaris.opengrok.history.RepositoryFactory;
+import org.opensolaris.opengrok.history.RepositoryInfo;
+import org.opensolaris.opengrok.index.IndexDatabase;
+import org.opensolaris.opengrok.index.Indexer;
+import static org.opensolaris.opengrok.util.IOUtils.removeRecursive;
+import org.opensolaris.opengrok.util.TestRepository;
+
+/**
+ *
+ * @author vkotal
+ */
+@ConditionalRun(condition = RepositoryInstalled.MercurialInstalled.class)
+// XXX need svn + git
```

Comment:
Simple way is to somehow trick it in `setUp` using Assume.assumeTrue which causes the test to be ignored when the condition is not met

The nice but more complicated is to implement the ConditionalRun annotation to be repeatable
https://docs.oracle.com/javase/tutorial/java/annotations/repeating.html may be a starter